File: backend/src/main/java/com/back/back9/domain/coin/dto/CoinDto.java
package com.back.back9.domain.coin.dto;

import com.back.back9.domain.coin.entity.Coin;

import java.time.LocalDateTime;

public record CoinDto(
        long id,
        String symbol,
        String koreanName,
        String englishName,
        LocalDateTime createdAt,
        LocalDateTime modifiedAt
) {
    public CoinDto(Coin coin){
        this(
                coin.getId(),
                coin.getSymbol(),
                coin.getKoreanName(),
                coin.getEnglishName(),
                coin.getCreatedAt(),
                coin.getModifiedAt()
        );
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/coin/dto/CoinDto.kt
package com.back.back9.domain.coin.dto

import com.back.back9.domain.coin.entity.Coin
import java.time.LocalDateTime

data class CoinDto(
    val id: Long?,
    val symbol: String,
    val koreanName: String?,
    val englishName: String?,
    val createdAt: LocalDateTime?,
    val modifiedAt: LocalDateTime?
) {
    constructor(coin: Coin) : this(
        coin.id,
        coin.symbol,
        coin.koreanName,
        coin.englishName,
        coin.createdAt,
        coin.modifiedAt
    )
}
